epi/src/file_storage.rs

```diff
@@ -1,26 +1,59 @@
 use std::{
     collections::HashMap,
     path::{Path, PathBuf},
+    sync::mpsc,
 };
 
 // ----------------------------------------------------------------------------
 
 /// A key-value store backed by a [RON](https://github.com/ron-rs/ron) file on disk.
 /// Used to restore egui state, glium window position/size and app state.
 pub struct FileStorage {
-    ron_filepath: PathBuf,
     kv: HashMap<String, String>,
     dirty: bool,
+    flush_handle: Option<std::thread::JoinHandle<()>>,
+    sender: mpsc::Sender<Message>,
+}
+
+enum Message {
+    DoFlush(HashMap<String, String>),
+    Terminate,
+}
+
+impl Drop for FileStorage {
+    fn drop(&mut self) {
+        self.sender.send(Message::Terminate).unwrap();
+        if let Some(handle) = self.flush_handle.take() {
+            handle.join().unwrap();
+        }
+    }
 }
 
 impl FileStorage {
     /// Store the state in this .ron file.
     pub fn from_ron_filepath(ron_filepath: impl Into<PathBuf>) -> Self {
         let ron_filepath: PathBuf = ron_filepath.into();
+        let cloned_filepath = ron_filepath.clone();
+        let (tx, rx): (mpsc::Sender<Message>, mpsc::Receiver<Message>) = mpsc::channel();
+        let flush_handle = Some(std::thread::spawn(move || loop {
+            let message = rx.recv().unwrap();
```

Comment:
```suggestion
        let flush_handle = Some(std::thread::spawn(move ||
            while let Ok(message) = rx.recv() {
```

less unwraps, and closes the thread when the sender drops. This also means we don't need `Message::Terminate`